From 9105e02b2aae442da106dffb038c784e6028d9df Mon Sep 17 00:00:00 2001
From: Ronan Abhamon <ronan.abhamon@vates.fr>
Date: Wed, 8 Jan 2025 11:12:13 +0100
Subject: [PATCH] Fix LVHDSR.load: set other_conf in cond-branch to prevent
 mypy error

Avoid:
```
drivers/LVHDSR.py:195: error: Item "None" of "Any | None" has no attribute "get"  [union-attr]
drivers/LVHDSR.py:196: error: Value of type "Any | None" is not indexable  [index]
```

Signed-off-by: Ronan Abhamon <ronan.abhamon@vates.fr>
---
 drivers/LVHDSR.py | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

Index: sm-3.2.12/drivers/LVHDSR.py
===================================================================
--- sm-3.2.12.orig/drivers/LVHDSR.py
+++ sm-3.2.12/drivers/LVHDSR.py
@@ -166,10 +166,11 @@ class LVHDSR(SR.SR):
         self.mdpath = os.path.join(self.path, self.MDVOLUME_NAME)
         self.provision = self.PROVISIONING_DEFAULT
 
-        self.other_conf = None
         has_sr_ref = self.srcmd.params.get("sr_ref")
         if has_sr_ref:
             self.other_conf = self.session.xenapi.SR.get_other_config(self.sr_ref)
+        else:
+            self.other_conf = None
 
         self.lvm_conf = None
         if self.other_conf:
